manifests/prepare.pp

```diff
@@ -16,11 +16,40 @@
     ensure => directory,
   }
 
-  file { $::puppet_agent::params::ssldir:
-    ensure  => directory,
-    source  => $::puppet_ssldir,
-    backup  => false,
-    recurse => true,
+  $ssl_dir = $::puppet_agent::params::ssldir
+  file { $ssl_dir:
+    ensure   => directory,
+    source   => $::puppet_ssldir,
+    backup   => false,
+    recurse  => false,
```

Comment:
@MikaelSmith updated the fact so its now "path_exists". Should hopefully be clearer what's happening now.